lib/versioner/files_counter.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2017 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package versioner
+
+import (
+	"path/filepath"
+	"sort"
+)
+
+// byLength is intended to sort empty dirs in reverse order
+type byLength []string
+
+func (s byLength) Len() int {
+	return len(s)
+}
+
+func (s byLength) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+func (s byLength) Less(i, j int) bool {
+	return len(s[i]) > len(s[j])
+}
+
+// filesCounter is intended to track empty directories
+type filesCounter struct {
```

Comment:
just `type filesCounter map[string]bool/` would do.

Reply:
Also on the very nitpicky level: This doesn't count files anymore, maybe emptyDirTracker (or something nicer) would be more appropriate.

Reply:
Agree